### ISSUE TITLE
Make channel order consistent between nn.Conv*  and nn.functional.conv*

### ISSUE THREAD

**Issue by tbenst** [pytorch/pytorch#9338]
Right now, nn.Conv* is `In channels x Out channels`, while nn.functional.conv* is `Out channels x In channels`. I'd be great if we could just pick one format for both. Personally, I prefer `In x Out` but I think consistency is more important. Since this would be breaking, seems like a good thing to fix before 1.0

**Comment by SsnL** [comment]
it's probably just doc error. they should take weight of same shape.

**Comment by tbenst** [comment]
The weight may be the same, but the function call is not. For functional, one writes `nn.Parameter(T.FloatTensor(64,8,5,5))` and for nn.conv2d the equivalent is `nn.Conv2d(8,64,(5,5),padding=self.dpad1)`

**Comment by SsnL** [comment]
oh i see. yeah I agree with @fmassa .

**Comment by soumith** [comment]
closing based on @fmassa 's comment. Agree with him.
